cypress/integration/GlossaryPopUps/GlossaryPopUps.js

```diff
@@ -0,0 +1,57 @@
+// <reference types="Cypress" />
+import { Given, And, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Cypress.on('uncaught:exception', (err, runnable) => {
+    // returning false here to Cypress from
+    // failing the test
+    return false;
+});
+
+
+When('user is clicking on definition for {string}', (termName) => {
+    cy.get('a.definition').first().should('be.visible');
+    cy.get('a.definition').contains(termName).click();
+    cy.wait(500);
+
+});
+
+Then('pop-up definition box is displayed for {string}', (termName) => {
+    cy.on('window:alert', (termName) => {
+        expect('div.modal__container .term').to.contains('termName');
+    })
+});
+
+And('definition box displays all {string} provided by glossary', (features) => {
+    const expText = features.split(',');
+    cy.on('window:alert', (expText) => {
+        expect('div.modal__container').to.contains('expText');
+        cy.get('div.modal__container').as('definition');
+
+        for (let i = 0; i <= expText.length; i++) {
+            switch (expText[i]) {
+                case 0:
+                    cy.get('@definition dd[class="definition"]').should('be.visible');
+                    break;
+                case 1:
+                    cy.get('@definition span[class="pronunciation"]').should('be.visible');
+                    break;
+                case 2:
+                    cy.get('@definition span[class="pronunciation"] a[class="CDR_audiofile"]').should('be.visible');
+                    break;
+                default:
+                    cy.get('@definition').find('img').as('image');
+                    cy.get('@image').should('have.attr', 'src', src);
```

Comment:
what is src and alt variables? You don't have it declared, how are you making an assertion?